pkg/syndesis/action/attach.go

```diff
@@ -0,0 +1,161 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/configuration"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Attach a detached Syndesis installation to the Syndesis custom resource
+type Attach struct {}
+
+
+func (a *Attach) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusAttaching)
```

Comment:
In which circumstance beside after an upgrade could it be that we lose the owner reference ? If its only after the upgrade, wouldn't it be more direct to just update the reference on all objects of any Kind with the label "app=syndesis" (see also my comment below about the labels) ?

Reply:
I've experienced issues on getting resources of "Any" kind with the client.. Will retry, but the Kind seems mandatory.